### PR TITLE
New module: Mindagap/duplicatefinder

### PR DESCRIPTION
**Pull request by FloWuenne**
## PR checklist

This PR adds the duplicate_finder function from Mindagap, which checks a spot table from Molecular Cartography for potential duplication of RNA spots and marks them. 

Closes ##3271 

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Comment by FloWuenne** [comment]
I still need to add test-data to test-datasets. Will take care of that later this weekend! Probably should also update the general tool description for mindagap, as now there will be two different subtools and there might be more in the future.

File: modules/nf-core/mindagap/duplicatefinder/main.nf [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    output:
+    tuple val(meta), path("*markedDups.txt"), emit: marked_dups_spots
+    path "versions.yml"                     , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
```
Comment (mashehu) [inline modules/nf-core/mindagap/duplicatefinder/main.nf:22]:
```suggestion
    def args   = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
```

File: modules/nf-core/mindagap/duplicatefinder/tests/main.nf.test [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+                    ]
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out.marked_dups_spots).match() },
+                { assert snapshot(process.out.versions).match("versions")}
```
Comment (sateeshperi) [inline modules/nf-core/mindagap/duplicatefinder/tests/main.nf.test:29]:
```suggestion
                { assert snapshot(process.out).match() }
```

Comment (sateeshperi) [reply modules/nf-core/mindagap/duplicatefinder/tests/main.nf.test:29]: if all your out channels generate consistent md5...recommend capturing all of it in one go to keep code concise

**Comment by FloWuenne** [comment]
Thanks for the suggestion @sateeshperi !

File: modules/nf-core/mindagap/duplicatefinder/tests/main.nf.test [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    tag "mindagap/duplicatefinder"
+
+    test("test_mindagap_duplicatefinder_spots") {
+
+        when {
+            process {
+                """
+                input[0] = [
+                    [ id:'test'], // meta map
+                    file('https://raw.githubusercontent.com/nf-core/test-datasets/molkart/test_data/input_data/spots.txt')
```
[the remark this answers is not in the document]
Comment (mashehu) [reply modules/nf-core/mindagap/duplicatefinder/tests/main.nf.test:19]: good idea. please make an issue on tools about it.

[the remark this answers is not in the document]
Comment (sateeshperi) [reply modules/nf-core/mindagap/duplicatefinder/tests/main.nf.test:19]: hmmm ya its going to be a different test data base I see...hmmm....let me think abt this...I mean its not a deal breaker as its still from test-datasets repo

[the remark this answers is not in the document]
Comment (FloWuenne) [reply modules/nf-core/mindagap/duplicatefinder/tests/main.nf.test:19]: Ok, so for now, can we leave it like this and please merge this PR @sateeshperi 😊🙏 ?

**Review by kbestak — APPROVED** [review]
Looks good!

**Pull request merged**